### ISSUE TITLE
Add Yarn in eslint --init

### ISSUE THREAD

**Issue by mateushnsoares** [eslint/eslint#13645]
---
name: "Add yarn support for --init"
about: Request a change that is not a bug fix, rule change, or new rule
title: 'Add yarn support for --init'
labels: enhancement, triage, core
assignees: ''

---

**The version of ESLint you are using.**

7.8.1

**The problem you want to solve.**

I'm 👎 on adding this in. Yarn is a popular package manager, but it's not an official one. There're a bunch of other package managers out there. If the user chose to use yarn as an NPM replacement (which is perfectly fine), they should be ready to do some tweaking by hands.

_Originally posted by @ilyavolodin in https://github.com/eslint/eslint/pull/9023#issuecomment-323495264_

The yarn is not official but is more used of NPM. And I don't see a problem on support Yarn ?? I really don't understand why closed  **[THIS PR](https://github.com/eslint/eslint/pull/9023)**

**Your take on the correct solution to problem.**

Add yarn and pnpm support.

**Are you willing to submit a pull request to implement this change?**

No

**Comment by anikethsaha** [comment]
Thanks for the issue. 

I think we can ask the user itself which package manager to use. That means one more prompt.

**Comment by kaicataldo** [comment]
Re-evaluating this now seems prudent, but please fill out the requested issue template. The current tone and state of this issue doesn't give us much room to have a discussion.

**Comment by mateushnsoares** [comment]
Sorry I use reply of the comment I will edit

**Comment by mateushnsoares** [comment]
@kaicataldo . I had edited, Sorry for the previous tone 😕

**Comment by kaicataldo** [comment]
Thank you. Adding a question about whether `yarn` or `npm` should be used in `--init` makes sense to me.

**Comment by snitin315** [comment]
Can I work on this issue?

**Comment by kaicataldo** [comment]
@snitin315 I'd recommend waiting until the team has reached consensus on this feature's inclusion so that there isn't any throwaway work. That being said, a PR can sometimes make things easier to evaluate!

**Comment by anikethsaha** [comment]
Is it required for this PR to go through RFC?

**Comment by kaicataldo** [comment]
@anikethsaha This seems like a small enough change to me that it wouldn't warrant an RFC.

**Comment by snitin315** [comment]
I will send a PR soon then.

**Comment by nzakas** [comment]
**TSC Summary:** This proposal seeks to add a question to `--init` asking if dependencies should be installed with npm or yarn.

**TSC Question:** Shall we accept this proposal?

**Comment by btmills** [comment]
We approved this change in today's TSC meeting. To reduce complexity, `--init` shouldn't try to guess the correct package manager. It should ask `Which package manager does your project use?` with `npm` as the default and `yarn` as an option. @snitin315 hopefully that's the answer you needed to finish up #13756! :rocket:

**Comment by snitin315** [comment]
@btmills thanks! I will update the corresponding PR.

**Comment by nzakas** [comment]
Reopening because of active PR.

**Comment by ashishbinu** [comment]
Hey , is anyone still working on this? It would be great if there is also `pnpm` support along with yarn.

If no one is working on this I can give it a shot.

**Comment by nzakas** [comment]
There is an active PR: https://github.com/eslint/eslint/pull/13756

**Comment by mdjermanovic** [comment]
We have decided to postpone `yarn` support until the `--init` functionality gets extracted into a separate package (https://github.com/eslint/rfcs/pull/58), for the reasons described in #13756.

**Comment by brettz9** [comment]
Given that the request also asked for pnpm (though I don't know if that part has been accepted), perhaps the issue can be renamed to ensure that support is also being tracked?

**Comment by nzakas** [comment]
Done

**Comment by sharedrory** [comment]
Just installed eslint and ran --init. After going through the prompts it starting downloading packages without my permission from npm. I use pnpm. I'm not saying add pnpm support (pnpm auto detects if other package managers install things and will prompt to fix it) rather don't auto start the installation. Uninstalled.

**Comment by mdjermanovic** [comment]
> Just installed eslint and ran --init. After going through the prompts it starting downloading packages without my permission from npm. I use pnpm. I'm not saying add pnpm support (pnpm auto detects if other package managers install things and will prompt to fix it) rather don't auto start the installation. Uninstalled.

@sharedrory can you please submit a bug report with steps that lead to downloading packages without your permission?

That is not intended behavior. I tried one path, and there was a step asking for confirmation to install packages with npm:

```js
√ How would you like to use ESLint? · style
√ What type of modules does your project use? · esm
√ Which framework does your project use? · react
√ Does your project use TypeScript? · No / Yes
√ Where does your code run? · browser
√ How would you like to define a style for your project? · guide
√ Which style guide do you want to follow? · airbnb
√ What format do you want your config file to be in? · JavaScript
Checking peerDependencies of eslint-config-airbnb@latest
The config that you've selected requires the following dependencies:

eslint-plugin-react@^7.21.5 eslint-config-airbnb@latest eslint@^5.16.0 || ^6.8.0 || ^7.2.0 eslint-plugin-import@^2.22.1 eslint-plugin-jsx-a11y@^6.4.1 eslint-plugin-react-hooks@^4 || ^3 || ^2.3.0 || ^1.7.0
? Would you like to install them now with npm? » No / Yes
```

**Comment by stereokai** [comment]
Any news about this? Looks like all the PRs/feats have been closed

**Comment by nzakas** [comment]
When there are updates they will be posted on this issue.
